lib/puppet/type/file/checksum_value.rb

```diff
@@ -0,0 +1,74 @@
+require 'puppet/util/checksums'
+
+module Puppet
+  Puppet::Type.type(:file).newproperty(:checksum_value) do
+    include Puppet::Util::Checksums
+    include Puppet::Util::Diff
+
+    desc "The checksum of the source contents. Only md5 and sha256 are supported when
+      specifying this parameter."
+
+    def insync?(is)
+      # If checksum_value is specified, only manage source.
+      # Content manages its own syncing... FOR NOW, duh duh duh
+      if resource.parameter(:source).nil?
+        return true
+      end
```

Comment:
I was confused by the comment "only manage source". If `checksum_value` is specified, but `source` is nil (so `checksum_value` must have been set explicitly with `content`) then we let `content` handle the `insync?`. If `source` is not nil, then `content` can't be set, and we handle the `insync?` logic below.